pkg/cloudprovider/fake/cloudprovider.go

```diff
@@ -66,7 +66,7 @@ func (c *CloudProvider) Create(ctx context.Context, nodeRequest *cloudprovider.N
 	// Labels
 	labels := map[string]string{}
 	for key, requirement := range instanceType.Requirements {
-		if requirement.Len() == 1 {
+		if requirement.Operator() == v1.NodeSelectorOpIn {
```

Comment:
Necessary since OS has multiple values but is not applied to this node. This caused a test to fail that was previously succeeding for bad reasons:
1. A node existed, but provisioning was not aware of cluster state
2. The node did not have the OS label -- for well known labels, this semantic means "anything goes"
3. The invariant that all well known labels are applied to nodes was not enforced, so the pod incorrectly scheduled.

Reply:
Ahh, it's in the fake only.